### PR TITLE
Embedder Service v0 with FlashBert

### PR DESCRIPTION
**Pull request by magdyksaleh**
Initialize:

```
docker run --gpus all --shm-size 1g -p 8080:80 -v $volume:/data \
    ghcr.io/predibase/lorax:main --model-id WhereIsAI/UAE-Large-V1
```

Embed:

```
curl 127.0.0.1:8080/embed \
-X POST \
-d '{
"inputs": "[INST] Natalia sold clips to 48 of her friends in April, and then she sold half as many clips in May. How many clips did Natalia sell altogether in April and May? [/INST]"
}' \
-H 'Content-Type: application/json'
```

Python:

```
client = Client(...)
resp = client.embed("What is the capital of France?")
resp.embeddings
```

### CODE REVIEW

**Comment by magdyksaleh** [comment]
curl 127.0.0.1:8080/embed \
    -X POST \
    -d '{
        "inputs": "[INST] Natalia sold clips to 48 of her friends in April, and then she sold half as many clips in May. How many clips did Natalia sell altogether in April and May? [/INST]"
    }' \
    -H 'Content-Type: application/json'

;)

File: clients/python/lorax/types.py [hunk 1]
```diff
@@ -336,6 +336,10 @@ class StreamResponse(BaseModel):
...
     sha: str
-
-    # Suppress pydantic warning over `model_id` field.
+    # Suppress pydantic warning over `model_id` field
     model_config = ConfigDict(protected_namespaces=())
+
+
+class EmbedResponse(BaseModel):
+    # Embeddings
+    embeddings: Optional[List[float]]
```
Comment (tgaddair) [inline clients/python/lorax/types.py:345]: Is this optional for embedding responses, or required?

File: server/lorax_server/models/__init__.py [hunk 2]
```diff
@@ -92,6 +92,10 @@ def get_model(
             dtypetrust_remote_code=trust_remote_code,
         )
 
+    if "WhereIsAI/UAE-Large-V1" in model_id:
```
Comment (tgaddair) [inline server/lorax_server/models/__init__.py:95]: Is there a way we can make this more generic to support any BERT based model?

File: server/lorax_server/models/types.py [hunk 3]
```diff
@@ -124,3 +124,19 @@ def to_pb(self) -> generate_pb2.Generation:
             next_tokens=self.next_tokens.to_pb(),
             generated_text=self.generated_text.to_pb() if self.generated_text is not None else None,
         )
+
+@dataclass
+class FlashBatch(ABC):
```
Comment (tgaddair) [inline server/lorax_server/models/types.py:129]: This seems a bit overloaded with the FlashBatchForCausalLM. Maybe this should be FlashBatchForEmbedding?

File: server/lorax_server/models/model.py [hunk 4]
```diff
@@ -86,6 +86,10 @@ def info(self) -> InfoResponse:
     @property
     def sliding_window_blocks(self) -> Optional[int]:
         return None
+    
+    @property
+    def supports_embeddings(self) -> bool:
```
Comment (tgaddair) [inline server/lorax_server/models/model.py:91]: Conversely, what happens when someone tries to use the BERT model for text generation?

**Review by tgaddair — COMMENTED** [review]
Looks great! Couple minor questions.

File: server/lorax_server/models/__init__.py (hunk 2 above)
Comment (magdyksaleh) [reply server/lorax_server/models/__init__.py:95]: Good call out

**Review by tgaddair — APPROVED** [review]

**Pull request merged**